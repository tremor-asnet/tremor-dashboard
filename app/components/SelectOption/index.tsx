interface SelectOptionProps {
  data: ListOptionProps[];
  onClickItem: () => void;
}

const SelectOption = ({ data, onClickItem }: SelectOptionProps) => (
  <>
    <ul className="absolute z-[1] w-[160px] right-0 shadow-tremor-cardImage dark:shadow-dark-select-option bg-secondary p-2 rounded-md dark:bg-dark-tremor-primary">
      {data.map(({ option, value }) => (
        <li
          key={option}
          value={value}
          className="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
          onClick={onClickItem}
          data-testid="option">
          Status: {option}
        </li>
      ))}
      <div className="h-px bg-gradient-select my-2 opacity-25 dark:bg-gradient-divider" />
      <li className="w-full text-tremor-default cursor-pointer text-attention px-4 py-[0.3rem] hover:bg-body hover:rounded-md min-h-[auto] dark:hover:bg-dark-secondary">
        Remove File
      </li>
    </ul>
  </>
);

export default SelectOption;
