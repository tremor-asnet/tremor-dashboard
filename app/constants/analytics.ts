/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
