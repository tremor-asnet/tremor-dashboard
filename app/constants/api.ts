import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
