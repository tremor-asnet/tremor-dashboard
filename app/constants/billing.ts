export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
