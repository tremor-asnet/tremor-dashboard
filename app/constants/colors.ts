type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
