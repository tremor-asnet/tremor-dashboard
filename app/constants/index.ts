export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./commons";
export * from "./status";
