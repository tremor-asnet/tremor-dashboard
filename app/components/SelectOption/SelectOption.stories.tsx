// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import { SelectOption } from "@/components";

// Constants
import { listOption } from "@/constants";

const meta = {
  title: "Components/SelectOption",
  tags: ["autodocs"],
  component: SelectOption,
} as Meta<typeof SelectOption>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => <SelectOption data={listOption} onClickItem={() => {}} />,
};
