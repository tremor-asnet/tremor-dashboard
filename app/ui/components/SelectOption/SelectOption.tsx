// Types
import { OptionType } from "@/types";
import { memo } from "react";

// Components
import { Button } from "..";

// Constants
import { VARIANT_BUTTON } from "@/constants";

interface SelectOptionProps {
  title: string;
  data: OptionType[];
  onSelectItem: (option: string, value: string) => void;
  onSelectRemove: () => void;
}

const SelectOption = ({
  data,
  title,
  onSelectItem,
  onSelectRemove,
}: SelectOptionProps) => {
  const handleSelectItem = (option: string, value: string) => () =>
    onSelectItem(option, value);

  return (
    <>
      <ul>
        {data.map(({ option, value }) => (
          <li
            className="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
            key={option}>
            <Button
              onClick={handleSelectItem(option, value)}
              value={value}
              variantTremor={VARIANT_BUTTON.LIGHT}
              data-testid={`item-filter-${value}`}
              additionalClass="text-secondary hover:text-primary dark:text-lighter dark:hover:text-lighter">
              {title}: {option}
            </Button>
          </li>
        ))}
      </ul>
      <div className="h-px bg-gradient-select my-2 opacity-25 dark:bg-gradient-divider" />
      <Button
        data-testid="remove-filter"
        additionalClass="w-full text-tremor-default hover:text-attention cursor-pointer justify-start text-attention dark:text-attention dark:hover:ttext-attention px-4 py-[0.3rem] hover:bg-body hover:rounded-md min-h-[auto] dark:hover:bg-dark-secondary text-left"
        onClick={onSelectRemove}
        variantTremor={VARIANT_BUTTON.LIGHT}>
        Remove Filter
      </Button>
    </>
  );
};

export default memo(SelectOption);
