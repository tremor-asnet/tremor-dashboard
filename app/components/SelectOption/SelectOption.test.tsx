import { render } from "@testing-library/react";

// Components
import SelectOption from ".";
import { listOption } from "@/constants";

const Props = {
  data: listOption,
  onClickItem: jest.fn(),
};

describe("Testing SelectOption component", () => {
  it("Should match snapshot", () => {
    const component = render(<SelectOption {...Props} />);
    expect(component).toMatchSnapshot();
  });
});
