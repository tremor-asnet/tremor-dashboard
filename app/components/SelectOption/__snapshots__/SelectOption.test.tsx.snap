// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SelectOption component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="absolute z-[1] w-[160px] right-0 shadow-tremor-cardImage dark:shadow-dark-select-option bg-secondary p-2 rounded-md dark:bg-dark-tremor-primary"
      >
        <li
          class="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
          data-testid="option"
          value="0"
        >
          Status: 
          Paid
        </li>
        <li
          class="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
          data-testid="option"
          value="2"
        >
          Status: 
          Refunded
        </li>
        <li
          class="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
          data-testid="option"
          value="1"
        >
          Status: 
          Canceled
        </li>
        <div
          class="h-px bg-gradient-select my-2 opacity-25 dark:bg-gradient-divider"
        />
        <li
          class="w-full text-tremor-default cursor-pointer text-attention px-4 py-[0.3rem] hover:bg-body hover:rounded-md min-h-[auto] dark:hover:bg-dark-secondary"
        >
          Remove File
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="absolute z-[1] w-[160px] right-0 shadow-tremor-cardImage dark:shadow-dark-select-option bg-secondary p-2 rounded-md dark:bg-dark-tremor-primary"
    >
      <li
        class="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
        data-testid="option"
        value="0"
      >
        Status: 
        Paid
      </li>
      <li
        class="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
        data-testid="option"
        value="2"
      >
        Status: 
        Refunded
      </li>
      <li
        class="w-full text-tremor-default cursor-pointer text-secondary px-4 py-[0.3rem] hover:bg-body hover:text-tremor-brand-subtle hover:rounded-md min-h-[auto] dark:text-dark-romance dark:hover:bg-dark-secondary"
        data-testid="option"
        value="1"
      >
        Status: 
        Canceled
      </li>
      <div
        class="h-px bg-gradient-select my-2 opacity-25 dark:bg-gradient-divider"
      />
      <li
        class="w-full text-tremor-default cursor-pointer text-attention px-4 py-[0.3rem] hover:bg-body hover:rounded-md min-h-[auto] dark:hover:bg-dark-secondary"
      >
        Remove File
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
